go/api/pubsub_message.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type PubSubMessage struct {
+	Message string
+	Channel string
+	Pattern Result[string]
+}
+
+func NewPubSubMessage(message, channel string) *PubSubMessage {
+	return &PubSubMessage{
+		Message: message,
+		Channel: channel,
+		Pattern: CreateNilStringResult(),
+	}
+}
+
+func NewPubSubMessageWithPattern(message, channel, pattern string) *PubSubMessage {
+	return &PubSubMessage{
+		Message: message,
+		Channel: channel,
+		Pattern: CreateStringResult(pattern),
+	}
+}
+
+func (msg *PubSubMessage) ToString() string {
+	jsonBytes, err := json.Marshal(msg)
+	if err != nil {
+		return ""
+	}
+	return string(jsonBytes)
+}
+
+// PushKind represents the type of push message received from the server
+type PushKind int
+
+// PushKind enum values - these match the numeric values from the Rust side
+const (
```

Comment:
See https://github.com/valkey-io/valkey-glide/pull/3417/files#r2021447845
cbindgen will create an enum for you

Reply:
changes applied